### PR TITLE
CI: Enable ASAN roce testing

### PR DESCRIPTION
**Pull request by ivankochin**
## What
Enable ASAN testing for `ucx_roce` setup.

### CODE REVIEW

File: buildlib/pr/main.yml [hunk 1]
```diff
@@ -432,6 +432,21 @@ stages:
...
+        container: ubuntu2204_ib
+        asan_check: yes
+    - template: tests.yml
+      parameters:
+        name: roce_proto_disable
+        demands: ucx_roce -equals yes
+        test_perf: 0
+        proto_enable: no
+        container: ubuntu2204_ib
+        asan_check: yes
```
Comment (yosefe) [inline buildlib/pr/main.yml:449]: IMO this is not needed - we can remove to reduce testing time
@brminich WDYT?

Comment (brminich) [reply buildlib/pr/main.yml:449]: maybe we can leave ASAN but remove valgrind for proto disable instead?

Comment (ivankochin) [reply buildlib/pr/main.yml]: Done.

**Review by yosefe — APPROVED** [review]

**Pull request merged**